spyderlib/widgets/editor.py

```diff
@@ -1917,6 +1925,41 @@ def dropEvent(self, event):
                 editor.insert_text( source.text() )
         event.acceptProposedAction()
 
+    #------ Tab reordering
+    # These methods are called from an event filter installed in 
+    # spyderlib\widets\tabs.py -> BaseTabs
+    def tab_pressed(self, event):
+        """ """
+        if event.button() == Qt.LeftButton:
+            self.moving_start_tab = self.tabbar.tabAt(event.pos())
+            self.moving_end_tab = None
+            self.tabbar.setCurrentIndex(self.moving_start_tab)
+            self.tabs.currentWidget().setFocus()
+
+    def tab_moved(self, event):
+        """ """
+        # If left button is not pressed then return
+        if not event.buttons() & Qt.LeftButton:
+            self.moving_end_tab = None
+            return
+        if not self.moving_tab:
+            QApplication.setOverrideCursor(Qt.ClosedHandCursor)
+            self.moving_tab = True
+        pos = event.pos()
+        self.moving_end_tab = self.tabbar.tabAt(pos)
+
+        if self.moving_start_tab is not self.moving_end_tab:
+            # This signal is connected to the Editor plugin and fires
+            # a method that reorders ALL openned editorstacks
+            self.sig_tabs_reordered.emit(self.moving_start_tab,
+                                         self.moving_end_tab)
+            self.moving_start_tab = self.moving_end_tab
+
+    def tab_released(self, event):
+        """ """
+        QApplication.restoreOverrideCursor()
+        self.moving_tab = False
+

```

Comment:
Do these methods really belong here? I mean, they seem to belong to `tabs.py` because they only deal with tabs, right?